File: 3394-check-if-grid-can-be-cut-into-sections/3394-check-if-grid-can-be-cut-into-sections.py
class Solution:
    def checkValidCuts(self, n: int, rectangles: List[List[int]]) -> bool:
        n = len(rectangles)
        
        def is_can(direction):
            rectangles.sort(key = lambda x : x[direction])

            last_end = rectangles[0][direction+2]
            cnt = 0

            for rectangle in rectangles[1:len(rectangles)]:
                # 시작점을 가져온다
                if rectangle[direction] >= last_end:
                    cnt += 1

                last_end = max(last_end, rectangle[direction+2])
            
            if cnt >= 2:
                return True

            return False
        
        return is_can(0) or is_can(1) 
            

File: 3394-check-if-grid-can-be-cut-into-sections/3394-check-if-grid-can-be-cut-into-sections.js
/**
 * @param {number} n
 * @param {number[][]} rectangles
 * @return {boolean}
 */
var checkValidCuts = function(n, rectangles) {
    const checkIsCan = (direct) => {
        rectangles.sort((a,b) => a[direct] - b[direct])
        let lastEnd = rectangles[0][direct+2]
        let cnt = 0

        for (let rectangle of rectangles){
            if (lastEnd <= rectangle[direct]){
                cnt += 1
            }
            lastEnd = Math.max(lastEnd, rectangle[direct+2])
        }

        return cnt >= 2 ? true : false

    }

    return checkIsCan(0) || checkIsCan(1)

};